pixeltable/tests/test_exprs.py

```diff
@@ -356,6 +358,28 @@ def test_arrays(self, test_tbl: catalog.Table) -> None:
         _ = t[t.array_col[:, 0]].show()
         print(_)
 
+    def test_type_cast(self, test_tbl: catalog.Table) -> None:
+        t = test_tbl
+        # For each column c2, ..., c7, we create a new column ci_as_str that casts it to
+        # a string, then check that each row is correctly converted
+        for col_id in range(2, 8):
+            col_name = f'c{col_id}'
+            as_str_col_name = f'c{col_id}_as_str'
+            t.add_column(**{as_str_col_name: getattr(t, col_name).astype(StringType())})
+            data = t.select(*[getattr(t, col_name), getattr(t, as_str_col_name)]).collect()
+            is_json_type = getattr(t, col_name).col_type.is_json_type()
```

Comment:
This isn't part of the user-visible API. We should probably make that clearer with private members.

You can use Table.column_types()[col_name] to get this.

Reply:
Hmm. I can change this I guess, but I'd argue it *should* be part of the user-visible API. Or at least, there should be some visible surrogate.

`column.my_type`

is a much more natural pattern than

`my_table.column_types()[col_name]`

i.e., as a user I'd like to think of the column as an object that knows about it's type.